mathesar/utils/forms.py

```diff
@@ -74,49 +90,45 @@ def create_form(form_def, user):
         server=database.server,
         schema_oid=form_def["schema_oid"],
         base_table_oid=form_def["base_table_oid"],
-        is_public=form_def.get("is_public", False),
+        access_role=access_role,
         header_title=form_def["header_title"],
         header_subtitle=form_def.get("header_subtitle"),
-        submit_role=submit_role,
         submit_message=form_def.get("submit_message"),
-        redirect_url=form_def.get("redirect_url"),
-        submit_label=form_def.get("submit_label")
+        submit_redirect_url=form_def.get("submit_redirect_url"),
+        submit_button_label=form_def.get("submit_button_label")
     )
     field_instances = [
         FormField(
-            key=field["key"],
-            attnum=field["attnum"],
+            key=field_def["key"],
             form=form_model,
-            index=field["index"],
-            kind=field["kind"],
-            label=field.get("label"),
-            help=field.get("help"),
-            readonly=field.get("readonly", False),
-            styling=field.get("styling"),
-            is_required=field.get("is_required", False),
+            index=field_def["index"],
+            label=field_def.get("label"),
+            help=field_def.get("help"),
+            kind=field_def["kind"],
+            column_attnum=field_def.get("column_attnum"),
+            related_table_oid=field_def.get("related_table_oid"),
+            fk_interaction_rule=field_def.get("fk_interaction_rule"),
             parent_field=None,
-            target_table_oid=field.get("target_table_oid"),
-            allow_create=field.get("allow_create", False),
-            create_label=field.get("create_label")
-        ) for field in form_def["fields"]
+            styling=field_def.get("styling"),
+            is_required=field_def.get("is_required", False),
+        ) for field_def, _parent in iterate_field_defs(form_def["fields"])
     ]
     created_fields = FormField.objects.bulk_create(field_instances)
-    fields_map = {field.key: field for field in created_fields}
+    created_fields_map = {field.key: field for field in created_fields}
     update_field_instances = []
-    for field in form_def["fields"]:
-        if field.get("parent_field_key"):
-            fields_map[field["key"]].parent_field = fields_map[field["parent_field_key"]]
-            update_field_instances.append(fields_map[field["key"]])
+    for field_def, parent_field_def in iterate_field_defs(form_def["fields"]):
+        if parent_field_def:
+            created_field = created_fields_map[field_def["key"]]
+            created_field.parent_field = created_fields_map[parent_field_def["key"]]
+            update_field_instances.append(created_field)
```

Comment:
If we were clever enough with BFS, we could probably do multiple bulk create steps. Not worth optimizing at the moment though.